portal-api/src/v1/controllers/amendments/patch-amendment.test.ts

```diff
@@ -0,0 +1,111 @@
+      // Arrange
+      const { req, res } = httpMocks.createMocks<PatchAmendmentRequest>({
+        params: { facilityId, amendmentId },
+        body: { update },
+        user,
+      });
+
+      // Act
+      await patchAmendment(req, res);
+
+      // Assert
+      expect(api.patchPortalFacilityAmendment).toHaveBeenCalledTimes(1);
+      expect(api.patchPortalFacilityAmendment).toHaveBeenCalledWith({
+        facilityId,
+        amendmentId,
+        update,
+        auditDetails: generatePortalAuditDetails(user._id),
+      });
+    });
+
+    it(`should respond with ${HttpStatusCode.Ok} and return the amendment`, async () => {
+      // Arrange
+      const mockPortalAmendmentResponse: PortalFacilityAmendmentWithUkefId = {
+        amendmentId,
+        facilityId,
+        type: AMENDMENT_TYPES.PORTAL,
+        ukefFacilityId: '123',
+        dealId,
+        createdAt: 1702061978881,
+        updatedAt: 1702061978881,
+        status: AMENDMENT_STATUS.IN_PROGRESS,
+      };
+
+      jest.mocked(api.patchPortalFacilityAmendment).mockResolvedValue(mockPortalAmendmentResponse);
+      const { req, res } = httpMocks.createMocks<PatchAmendmentRequest>({
+        params: { facilityId, amendmentId },
+        body: { update },
+        user,
+      });
+
+      // Act
+      await patchAmendment(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+      expect(res._getData()).toEqual(mockPortalAmendmentResponse);
+    });
+
+    it('should return an error when there is an API error', async () => {
+      const testErrorStatus = HttpStatusCode.Forbidden;
+      const testApiErrorMessage = 'test api error message';
+      jest.mocked(api.patchPortalFacilityAmendment).mockRejectedValue(new TestApiError({ status: testErrorStatus, message: testApiErrorMessage }));
+
+      // Arrange
+      const { req, res } = httpMocks.createMocks<PatchAmendmentRequest>({
+        params: { facilityId, amendmentId },
+        body: { update },
+        user,
+      });
+
+      // Act
+      await patchAmendment(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(testErrorStatus);
+      expect(res._getData()).toEqual({ message: `Failed to update the amendment: ${testApiErrorMessage}`, status: testErrorStatus });
+    });
+
+    it('should return an error when there is a general error', async () => {
+      jest.mocked(api.patchPortalFacilityAmendment).mockRejectedValue(new Error('Some error'));
+
+      // Arrange
+      const { req, res } = httpMocks.createMocks<PatchAmendmentRequest>({
+        params: { facilityId, amendmentId },
+        body: { update },
+        user,
+      });
+
+      // Act
+      await patchAmendment(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.InternalServerError);
+      expect(res._getData()).toEqual({ message: 'Failed to update the amendment', status: HttpStatusCode.InternalServerError });
```

Comment:
Add an assertion for console.error please and `expect(api.patchPortalFacilityAmendment).toHaveBeenCalledTimes(0)`;

Reply:
Done